### ISSUE TITLE
unicorn/prefer-top-level-await incorrectly flags any call to .then as a promise

### ISSUE THREAD

**Issue by jessekelly881** [sindresorhus/eslint-plugin-unicorn#1942]
Unicorn seems to be flagging any call to .then() as a promise when it is not. 

unicorn/prefer-top-level-await

This code should not be flagged.
```js
import * as R from 'fp-ts-routing';
const test = R.lit('test').then(R.end)
```

**Comment by sindresorhus** [comment]
`.then` should not be used for anything other than promises as it comes with some magic behavior: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/7d804e3ce8ecb9d51b16d558eeae6dc12a81ec4c/docs/rules/no-thenable.md

**Comment by sindresorhus** [comment]
There's no way for the rule to know it's not a promise from just static checking.

**Comment by jessekelly881** [comment]
Oh interesting. I was not aware of this. That makes sense though.
